### PR TITLE
FEAT: Make tifffile read `resolution` metadata

### PR DESCRIPTION
**Pull request by FirefoxMetzger**
Closes: #731 

This PR implements the ability for tifffile to report the `resolution_unit` tag. This was documented before but appears to have not been implemented. I took the liberty of changing its behavior to match the behavior of the ResolutionUnit tag in the TIFF spec.

Further, this PR implements a new metadata item called `resolution` that matches the behavior of the `resolution` metadata item in write. aka: the ability to read dpi/dpc if specified in the TIFF.

### CODE REVIEW

File: imageio/plugins/tifffile.py [hunk 1]
```diff
@@ -42,8 +42,20 @@
...
     'contig': last dimension contains samples.
     'planar': third last dimension contains samples.
-resolution_unit : (float, float) or ((int, int), (int, int))
-    X and Y resolution in dots per inch as float or rational numbers.
+resolution_unit : int
+    The resolution unit stored in the TIFF tag. Usually 1 means no/unknown unit,
+    2 means dpi (inch), 3 means dpc (centimeter).
+resolution : ((int, int), (int, int), str)
+    A tuple formatted as (X_resolution, Y_resolution, unit). Both resolutions
+    are given as rational numbers, i.e. as a (numerator, denominator) tuple. Unit
```
Comment (almarklein) [inline imageio/plugins/tifffile.py:50]: Would it make more sense to return these as floats, or as [real rational numbers](https://docs.python.org/3/library/fractions.html)?

File: tests/test_tifffile.py [hunk 2]
```diff
@@ -144,6 +144,34 @@ def test_imagej_hyperstack(tmp_path):
...
+    dpi = (100, 200)
+
+    writer = imageio.get_writer(tmp_path / "test.tif")
+    writer.append_data(data, dict(resolution=dpi))
+    writer.close()
+
+    read_image = iio.imread(tmp_path / "test.tif")
+
+    assert read_image.meta["resolution"] == (*dpi, "INCH")
+    assert read_image.meta["resolution_unit"] == 2
```
Comment (almarklein) [inline tests/test_tifffile.py:160]: I'm confused, here the numbers are not rationals?

Comment (FirefoxMetzger) [reply tests/test_tifffile.py:160]: True. Yesterday, it seemed like a good idea to reduce a rational like `(200, 1)` to `200`, i.e., if the denominator is 1, drop it. Today, I'm not so sure anymore how good this idea actually is, considering that it makes the return type inconsistent.

What do you think?

Comment (almarklein) [reply tests/test_tifffile.py:160]: I think we should at least be consistent. So either we always return `(int, int)` or we always return `float`. My gut says floats but I also see the value in just passing what tifffile returns. Your call :D

Comment (almarklein) [reply tests/test_tifffile.py:160]: I cannot imagine that the precision matters *that* much. I think I'm leaning towards floats too; the int-tuple is confusing.

Comment (FirefoxMetzger) [reply tests/test_tifffile.py]: lets go with `float` then. I made the necessary changes.

**Review by almarklein — APPROVED** [review]

**Pull request merged**